boxes/generators/platonic.py

```diff
@@ -0,0 +1,174 @@
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+from boxes.edges import FingerJointBase, FingerJointEdge
+
+from math import sin, pi
+
+class UnevenJointBase(FingerJointBase):
+    def calcFingers(self, length, bedBolts):
+        space, finger = self.settings.space, self.settings.finger
+        fingers = int((length - (self.settings.surroundingspaces - 1) * space) //
+                      (space + finger))
+        if not finger:
+            fingers = 0
+        if bedBolts:
+            fingers = bedBolts.numFingers(fingers)
+        leftover = length - fingers * (space + finger)  # we don't another space
+
+        if fingers <= 0:
+            fingers = 0
+            leftover = length
+
+        return fingers, leftover
+
+class UnevenFingerJointEdge(FingerJointEdge, UnevenJointBase):
+    """Uneven finger joint edge """
+    char = 'u'
+    description = "Uneven Finger Joint"
+    positive = True
+
+    def __call__(self, length, bedBolts=None, bedBoltSettings=None, **kw):
+        # copied from original
+
+        positive = self.positive
+
+        s, f, thickness = self.settings.space, self.settings.finger, self.settings.thickness
+
+        p = 1 if positive else -1
+
+        fingers, leftover = self.calcFingers(length, bedBolts)
+
+        if not positive:
+            play = self.settings.play
+            f += play
+            s -= play
+            leftover -= play
+
+        self.edge(leftover, tabs=1)  # Whole point of this class
+
+        l1,l2 = self.fingerLength(self.settings.angle)
+        h = l1-l2
+
+        d = (bedBoltSettings or self.bedBoltSettings)[0]
+
+        for i in range(fingers):
+            if i != 0:
+                if not positive and bedBolts and bedBolts.drawBolt(i):
+                    self.hole(0.5 * s,
+                              0.5 * self.settings.thickness, 0.5 * d)
+
+                if positive and bedBolts and bedBolts.drawBolt(i):
+                    self.bedBoltHole(s, bedBoltSettings)
+                else:
+                    self.edge(s)
+
+            if positive and self.settings.style == "springs":
+                self.polyline(
+                    0, -90 * p, 0.8*h, (90 * p, 0.2*h),
+                    0.1 * h, 90, 0.9*h, -180, 0.9*h, 90,
+                    f - 0.6*h,
+                    90, 0.9*h, -180, 0.9*h, 90, 0.1*h,
+                (90 * p, 0.2 *h), 0.8*h, -90 * p)
+            else:
+                self.polyline(0, -90 * p, h, 90 * p, f, 90 * p, h, -90 * p)
+
+        self.edge(0, tabs=0)  # Whole point of this class
```

Comment:
This is not doing anything. Which means you are also loosing the second tab on this edge. But that should not be a big issue and also can be added later to the last line of the last polyline() above.